dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3424,6 +3424,36 @@ def test_pyarrow_dataset_filter_partitioned(tmpdir, split_row_groups):
     )
 
 
+def test_pyarrow_dataset_filter_on_partitioned(tmpdir, engine):
+    # See: https://github.com/dask/dask/issues/9246
+    ddf = dd.from_map(
+        lambda x: pd.DataFrame({"a": [x], "b": [x]}),
+        "abcdefg",
+    )
+    ddf["b"] = ddf["b"].astype("category")
+    ddf.to_parquet(tmpdir, engine=engine, partition_on=["b"])
+
+    # Check List[Tuple] filters
+    read_ddf = dd.read_parquet(
+        tmpdir,
+        engine=engine,
+        filters=[("b", "==", "c")],
+    )
+    assert read_ddf.npartitions == 1
+    assert_eq(read_ddf["a"], ddf.partitions[2]["a"])
+
+    # Check List[List[Tuple]] filters
+    # for "pyarrow" engine only
+    if engine == "pyarrow":
```

Comment:
Why only check the pyarrow engine here? Does fastparquet not support this format for filters?

Reply:
Yes exactly. @martindurant  may be able to clarify exactly what fastparquet supports, but I found [this docstring note](https://github.com/dask/dask/blob/3456d42e5642700f4cb2948f5b0d0e3f37306a82/dask/dataframe/io/parquet/core.py#L227) to be true.

Reply:
Gotcha -- thanks for clarifying!

Reply:
We made an effort to support exactly the syntax pyarrow uses (fastparquet had it first!), so if there's a case we're missing, we can fix it.

Reply:
Thanks @martindurant  - You are correct that `fastparquet` itself should be able to support DNF filters. It looks like the problem is that we are relying on `fastparquet.api.filter_out_cats` for partitioned columns, and Dasks `apply_filters` utility for everything else. Neither of these specific functions support the DNF format.

Perhaps it is time to let the "fastparquet" engine take complete ownership over filtering (like we do for "pyarrow").  We can just filter all row-groups with `fastparquet.api.filter_row_groups` (**and** use row-wise filtering), rather than relying on the obsolete `apply_filters` utility. Does this sound reasonable?

That said - This change should certainly be a seperate PR.